app_feup/lib/view/Widgets/bug_report_form.dart

```diff
@@ -169,83 +186,113 @@ class BugReportFormState extends State<BugReportForm> {
+    return Container(
         child: ElevatedButton(
-      onPressed: () {
-        if (_formKey.currentState.validate() && !_isButtonTapped) {
-          submitBugReport();
-        }
-      },
+      style: buttonStyle,
+      onPressed: !_isConsentGiven
+          ? null
+          : () {
+              if (_formKey.currentState.validate() && !_isButtonTapped) {
+                if (!FocusScope.of(context).hasPrimaryFocus) {
+                  FocusScope.of(context).unfocus();
+                }
+                submitBugReport();
+              }
+            },
       child: Text(
         'Enviar',
         style: TextStyle(color: Colors.white, fontSize: 20.0),
       ),
-      style: ElevatedButton.styleFrom(
-        padding: EdgeInsets.symmetric(vertical: 10.0),
-        primary: Theme.of(context).primaryColor,
-      ),
     ));
   }
 
-  void submitBugReport() {
+  void submitBugReport() async {
     setState(() {
       _isButtonTapped = true;
     });
 
+    final String title = titleController.text;
     final String bugLabel = bugDescriptions[_selectedBug] == null
         ? 'Unidentified bug'
         : bugDescriptions[_selectedBug].item2;
-    final Map data = {
-      'headline6': titleController.text,
-      'body': descriptionController.text,
-      'labels': [_issueLabel, bugLabel]
-    };
-
-    final url = _postUrl + '?access_token=' + ghToken;
-    http
-        .post(url.toUri(),
-            headers: {'Content-Type': 'application/json'},
-            body: json.encode(data))
-        .then((http.Response response) {
-      final int statusCode = response.statusCode;
-
-      String msg;
-      if (statusCode < 200 || statusCode > 400) {
-        Logger().e('Error ' + statusCode.toString() + ' while posting bug');
-        msg = 'Ocorreu um erro no envio';
-      } else {
-        Logger().i('Successfully submitted bug report.');
-        msg = 'Enviado com sucesso';
-
-        clearForm();
-
-        Navigator.pop(context);
-        setState(() {
-          _isButtonTapped = false;
-        });
-      }
-      
-
-      FocusScope.of(context).requestFocus( FocusNode());
-      displayBugToast(msg);
-      setState(() {
-        _isButtonTapped = false;
-      });
-    }).catchError((error) {
-      Logger().e(error);
-      FocusScope.of(context).requestFocus( FocusNode());
-
-      final String msg =
-          (error is SocketException) ? 'Falha de rede' : 'Ocorreu um erro';
-      displayBugToast(msg);
-      setState(() {
-        _isButtonTapped = false;
+
+    String toastMsg;
+    try {
+      final String sentryDescription = emailController.text == ''
+          ? descriptionController.text
+          : descriptionController.text + '\nContact: ' + emailController.text;
+      final sentryId = await Sentry.captureMessage(
+          bugLabel + ': ' + title + '\n' + sentryDescription);
+
+      final String gitHubDescription =
+          descriptionController.text + '\n' + _sentryLink + sentryId.toString();
```

Comment:
It would be wise to have some sort of explanation before the link to sentry, like "For more information, head to http://...". wdyt?

Reply:
I agree!